frontend/sw360-portlet/src/main/webapp/html/admin/todos/view.jsp

```diff
@@ -0,0 +1,139 @@
+
+<portlet:renderURL var="addTodoURL">
+    <portlet:param name="<%=PortalConstants.PAGENAME%>" value="<%=PortalConstants.PAGENAME_ADD%>" />
+</portlet:renderURL>
+
+<jsp:useBean id="todoList" type="java.util.List<org.eclipse.sw360.datahandler.thrift.licenses.Todo>" scope="request"/>
+
+<link rel="stylesheet" href="<%=request.getContextPath()%>/webjars/datatables.net-buttons-bs/css/buttons.bootstrap.min.css"/>
+<link rel="stylesheet" href="<%=request.getContextPath()%>/css/dataTable_Siemens.css">
+<link rel="stylesheet" href="<%=request.getContextPath()%>/css/sw360.css">
+
+<div id="header"></div>
+<p class="pageHeader">
+    <span class="pageHeaderBigSpan">TODOs</span> <span class="pageHeaderSmallSpan">(${todoList.size()})</span>
+    <span class="pull-right">
+        <input type="button" class="addButton" onclick="window.location.href='<%=addTodoURL%>'" value="Add Todo">
+    </span>
+</p>
+
+<div id="searchInput" class="content1">
+    <%@ include file="/html/utils/includes/quickfilter.jspf" %>
+</div>
+<div id="todoTableDiv" class="content2">
+    <table id="todoTable" cellpadding="0" cellspacing="0" border="0" class="display">
+        <col width="25%">
+        <col width="55%">
+        <col width="10%">
+        <col width="10%">
+        <tfoot>
+        <tr>
+            <th colspan="4"></th>
+        </tr>
+        </tfoot>
+    </table>
+</div>
+
+<link rel="stylesheet" href="<%=request.getContextPath()%>/webjars/jquery-confirm2/dist/jquery-confirm.min.css">
+
+<%--for javascript library loading --%>
+<%@ include file="/html/utils/includes/requirejs.jspf" %>
+<script>
+    AUI().use('liferay-portlet-url', function () {
+
+        require(['jquery', 'utils/includes/quickfilter', 'modules/confirm', /* jquery-plugins: */ 'datatables.net', 'datatables.net-buttons', 'datatables.net-buttons.print', 'jquery-confirm'], function($, quickfilter, confirm) {
+
+            var todoTable = createTodoTable();
+            quickfilter.addTable(todoTable);
+
+            // register event handlers
+            $('#todoTable').on('click', 'img.delete', function (event) {
+                var data = $(event.currentTarget).data();
+                deleteTodo(data.id, data.title);
+            });
+
+
+            function createTodoTable() {
+                var todosTbl,
+                    result = [];
+
+                <core_rt:forEach items="${todoList}" var="todo">
+                result.push({
+                    "DT_RowId": "${todo.id}",
+                    "0": "${todo.title}",
+                    "1": '<pre style="background-color:#ffffff;border:none"><sw360:out value="${todo.text}" stripNewlines="false" jsQuoting="true"/></pre>',
+                    "2": "<input type='checkbox' disabled='true' <core_rt:if test="${todo.validForProject}">checked</core_rt:if>>",
+                    "3": "<img class='delete' src='<%=request.getContextPath()%>/images/Trash.png' data-id='${todo.id}'  data-title='${todo.title}' alt='Delete' title='Delete'>"
```

Comment:
If I am correct, then we have the logic that an action icon is not shown at all if the current user has not enough permissions (at least in a few places). What do you think about such an approach?

Reply:
That was going to be my initial approach but given that the TODOs portlet is part of the admin panel if the user can see that then already is an admin. If this sounds incorrect please clarify.

Reply:
Ah, the portlet definition in `frontend/sw360-portlet/src/main/webapp/WEB-INF/portlet.xml` contains the role restriction. Then it might be fine (even though I could not completely clarify if this role is only used for permissions to add the portlet to pages).

Btw: just the page we are using the portlet on should not be a sufficient criteria because other users might pick the portlet on completely different pages with different access rights. Our page tree is just a suggestion.
But this brought up the problem that your new page does not have the proper rights. It seems that page permission in liferay are not inherited. So please set the proper permissions on the new Todo page before generating the new .lar file.

Reply:
Very well, will do.